_inc/lib/class.core-rest-api-endpoints.php

```diff
@@ -2956,6 +2985,111 @@ private static function core_get_plugins() {
 		return array();
 	}
 
+
+	/**
+	 * Get third party plugin API keys.
+	 *
+	 * @param WP_REST_Request $request {
+	 *     Array of parameters received by request.
+	 *
+	 *     @type string $slug Plugin slug with the syntax 'plugin-directory/plugin-main-file.php'.
+	 * }
+	 */
+	public static function get_api_key( $request ) {
+		$service = $request['service'];
+		$option  = self::key_for_api_service( $service );
```

Comment:
I know `Jetpack_Options` restricts what option names we can pass, but it still feels dangerous not to restrict the `service` values here (and in the other endpoints).

For example, if some future code added a `secret_api_key` option, we could get/delete/update it via the API: `jetpack/v4/api-key/secret`.

Reply:
`esc_url_raw()` change made in https://github.com/Automattic/jetpack/pull/10393/commits/8274b5e211418496252ea6a58c6882d06c9c5ba9

Reply:
> it still feels dangerous not to restrict the service values here (and in the other endpoints).

Very good point. I added a check in https://github.com/Automattic/jetpack/pull/10393/commits/18cfd08d0ce4c3e4030303dec539e5f5db3fdeae